### PR TITLE
Show ApplicationBannerCard all the time and add /application/last endpoint

### LINKED ISSUE #122: /application/last endpoint & Display banner forever
From now on, we should display the ApplicationBannerCard (the one that show the status of your application) all the time.
At this time, it is not visible since there are no active application periods, although it could display important information like handed-out-pass' statuses.
Ask @FearsomeRover about this issue!

### PR DESCRIPTION
**Pull request by EasySouls**
I created the endpoint which works similarly than the existing /application/my endpoint, only that it doesn't take the current application period into consideration.

On the frontend now the banner has 4 states:
- If the user is not logged in, there are no currently active periods and the user has never submitted an application it is not shown.
- If there is a submitted application for the active period, the card is shown, just as before
- If the user has not submitted an application, but there is an active period, the same button is shown which leads to the application form, just as before
- If there is no active period, but the user has a previous application, the card is shown in the same form like with an active period, only now it is presented to the user that in which period was the card submitted.

Closes #122

### CODE REVIEW

**Comment by EasySouls** [comment]
@FearsomeRover Egyelőre a typescript errort jelez, de különben működne, csak direkt nem javítottam ki, mert észrevettem valamit. Az `/application/my` és az új endpoint is egy `Application`-t ad vissza ts szerint. 
Viszont az eddig is létező `useCurrentApplication()` hook pedig a `ApplicationEntityWithPeriod` típust rendeli hozzá. Szóval ts szerint létezik period mezője, de mivel a backendről csak egy `Application`-t kap vissza, ebből csak undefined lesz.
A kérdésem az, hogy írjam át a backendes kódot, hogy includeolja a period entity-t is, vagy az frontenden igazából nem is kell, és a `useCurrentApplication()` és az új `useLastApplication()` kaphatnak sima `Application`-t?

File: apps/backend/src/application/application.service.ts [hunk 1]
```diff
@@ -128,6 +135,35 @@ export class ApplicationService {
...
+   * @throws {NotFoundException} - if the user has no applications
+   */
+  async getLastUserApplication(user: User): Promise<Application> {
+    try {
+      return await this.prisma.application.findFirstOrThrow({
+        where: {
+          userId: user.authSchId,
+        },
+        orderBy: {
+          createdAt: 'desc',
```
Comment (mozsarmate) [inline apps/backend/src/application/application.service.ts:152]: lehetne applicationPeriodEnd alapján

File: apps/frontend/src/components/ui/ApplicationBannerCard.tsx [hunk 2]
```diff
@@ -35,8 +37,56 @@ export default function ApplicationBannerCard() {
...
+      </Card>
+    );
+  } else if (lastApplication.data) {
+    return (
+      <Card className='w-full'>
+        <CardHeader className='md:flex-row max-md:flex-col w-full justify-between gap-2 md:items-start'>
+          <div className='flex flex-col gap-4 justify-start'>
+            <CardTitle>Eddigi jelentkezésed</CardTitle>
+            <p>
+              Jelenleg nincsen jelentkezési időszak, de itt láthatod a(z){' '}
```
Comment (mozsarmate) [inline apps/frontend/src/components/ui/ApplicationBannerCard.tsx:78]: szerintem ezt kivehetjük, amúgy is straight-forward

**Review by mozsarmate — APPROVED** [review]
Patika munka, nyomoood!

**Comment by EasySouls** [comment]
I also refactored the types regarding the applications. Since the currentApplication wasn't using the applicationPeriod's data, I changed the swr function's return type to be simply ApplicationEntity.
And changed the lastApplication's return type to be ApplicationEntityWithPeriod, since it might be useful to get that periods information in which the last application was submitted in.
In the backend controller I corrected the return type as well, but it is quite nasty looking, and since it uses Prisma's types, it doesn't show up in the OpenAPI specs, so it might be better to change it to our own dto in the future.

**Comment by EasySouls** [comment]
What I meant to say is that before the changes, we could get the currentApplication's period data in the frontend, but since the backend doesn't include it in the response, it would result in an error. 
And now we can typesafely get the period data using the `useLastApplication()` hook.

**Review by mozsarmate — APPROVED** [review]

**Comment by mozsarmate** [comment]
Looks good to me, nice work!

**Pull request merged**